src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -0,0 +1,224 @@
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [-1, 3],
+                    },
+                ),
+            ],
+        )
+
+    def _get_zocalo_connection(self):
+        zc = zocalo.configuration.from_file()
+        zc.activate_environment(self.zocalo_environment)
+
+        transport = lookup("PikaTransport")()
+        transport.connect()
+        return transport
+
+    def _subscribe_to_results(self):
+        transport = self._get_zocalo_connection()
+
+        def _receive_result(
+            rw: workflows.recipe.RecipeWrapper, header: dict, message: dict
+        ) -> None:
+            LOGGER.info(f"Received {message}")
+            recipe_parameters = rw.recipe_step["parameters"]  # type: ignore # this rw is initialised with a message so recipe step is not None
+            LOGGER.info(f"Recipe step parameters: {recipe_parameters}")
+            transport.ack(header)
+
+            results = message.get("results", [])
+            call_in_bluesky_event_loop(self._raw_results_received.put(results))
+
+        subscription = workflows.recipe.wrap_subscribe(
+            transport,
+            self.channel,
+            _receive_result,
+            acknowledgement=True,
+            allow_non_recipe_messages=False,
+        )
+        LOGGER.info(f"Made zocalo queue subscription: {bool(subscription)}.")
+
+
+ZOCALO_READING_PLAN_NAME = "zocalo reading"
+
+
+def trigger_wait_and_read_zocalo(zocalo: ZocaloResults):
+    """A minimal utility plan which will wait for analysis results to be returned from
+    Zocalo, and bundle them in a reading."""
+
+    yield from bps.create(ZOCALO_READING_PLAN_NAME)
+    LOGGER.info("Running zocalo device .trigger()")
+    yield from bps.trigger(zocalo, wait=True)
+    yield from bps.read(zocalo)
+    yield from bps.save()
+
+
+def get_processing_results(
+    zocalo: ZocaloResults,
+) -> Generator[Any, Any, Union[Tuple[np.ndarray, np.ndarray], Tuple[None, None]]]:
+    """A minimal plan which will extract the top ranked xray centre and crystal bounding
+    box size from the zocalo results."""
+    LOGGER.info("Getting zocalo processing results.")
+    centres_of_mass = yield from bps.rd(zocalo.centres_of_mass, default_value=[])  # type: ignore
+    LOGGER.info(f"Centres of mass: {centres_of_mass}")
+    centre_of_mass = (
+        None
+        if len(centres_of_mass) == 0  # type: ignore
+        else centres_of_mass[0] - np.array([0.5, 0.5, 0.5])  # type: ignore
+    )
+    LOGGER.info(f"Adjusted top centring result: {centre_of_mass}")
+    bbox_sizes = yield from bps.rd(zocalo.bbox_sizes, default_value=[])  # type: ignore
+    LOGGER.info(f"Bounding box sizes: {centres_of_mass}")
+    bbox_size = None if len(bbox_sizes) == 0 else bbox_sizes[0]  # type: ignore
+    LOGGER.info(f"Top bbox size: {bbox_size}")
+    return centre_of_mass, bbox_size
```

Comment:
Nit: I think this is overly logged for production code, can we reduce them down a bit?